benchmarks/run_benchmarks.py

```diff
@@ -188,7 +193,25 @@
 
 start_dir = os.getcwd()
 
-code_folder = os.path.join(os.path.dirname(__file__), 'benchmark_code')
+code_folder = os.path.join(os.path.dirname(__file__), 'tests')
+
+def run_process(cmd: "List[str]", time_compilation: "bool"=False):
+    if not time_compilation:
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                universal_newlines=True)
+        out, err = p.communicate()
+        return p, out, err, 0.0
+
+    usage_start = resource.getrusage(resource.RUSAGE_CHILDREN)
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+            universal_newlines=True)
+    out, err = p.communicate()
+    usage_end = resource.getrusage(resource.RUSAGE_CHILDREN)
+    cpu_time = sum([
```

Comment:
Please double-check that the number returned here is correct and matches the use of `time` on your machines, as I am not able to confidently test it on my machines.

Reply:
For some reason, I'm struggling to persuade this to trigger, even if I remove the condition on the commit message

Reply:
Excluding the first case which may just be an outlier, the results seem to be roughly equivalent:
```
10c10
< Compilation CPU time :  1.5860259999999997
---
> Compilation CPU time :  2.03
20c20
< Compilation CPU time :  2.2057729999999993
---
> Compilation CPU time :  2.23
30c30
< Compilation CPU time :  2.3367850000000003
---
> Compilation CPU time :  2.3099999999999996
40c40
< Compilation CPU time :  2.2534690000000017
---
> Compilation CPU time :  2.2399999999999998
50c50
< Compilation CPU time :  2.5952830000000002
---
> Compilation CPU time :  2.59
60c60
< Compilation CPU time :  2.645474999999996
---
> Compilation CPU time :  2.66
70c70
< Compilation CPU time :  3.255339999999999
---
> Compilation CPU time :  3.24
80c80
< Compilation CPU time :  2.1001440000000065
---
> Compilation CPU time :  2.1100000000000003
90c90
< Compilation CPU time :  2.1116199999999976
---
> Compilation CPU time :  2.11
```

Reply:
Would you like me to investigate the first case?

Reply:
If you have any ideas about what could be causing this, I would be interested.

Now the tests have run on the main branch it seems that this is not limited to one case.
As you can see here: https://github.com/pyccel/pyccel-benchmarks/commit/2b6427a450b25909704f6e16ba2a7d8e65547868 , the times seem to have increased by 1s for most cases. As this is the case for both pyccel and pythran I don't know if it is a problem. It doesn't change the comparison in the paper, but the figures probably need updating.

I suspect the difference is either caused by an overhead due to `Popen` or, more likely, by a difference between real, user, and system time.

I suspect that the current results provide a better comparison with numba (whose compilation time is calculated from python).

cc: @yguclu

Reply:
Off hand I don't have any concrete ideas. I don't object to the change being reverted and the PR being re-opened - it's certainly not my intent to derail things by breaking your software. :)

Some speculation:

My first thought is that the units reported by `time` and by `resource.ru_*time` might not be the same (seconds, milliseconds, ticks, etc). For example, on my system, `time` reports in this format, while the original regex is looking for something like `0.0user 0.0system`.

```sh
$ time sleep 5

real    0m5.003s
user    0m0.002s
sys     0m0.000s
```

> I suspect the difference is either caused by an overhead due to Popen

I don't think this is the case because `resource`, when used with `RUSAGE_CHILDREN`, should only report time spent by the child process. That is, the time measurement does not include the spent spent to marshal the calls through `Popen` and Python. See [getusage(2)](https://man7.org/linux/man-pages/man2/getrusage.2.html) and [resource.getusage](https://docs.python.org/3/library/resource.html#resource.RUSAGE_CHILDREN).

> or, more likely, by a difference between real, user, and system time.

Possibly! The times are close enough that it might be difficult to know with certainty, however I would have guessed that _not_ using `time` would result in faster numbers if this were the case.

> I suspect that the current results provide a better comparison with numba (whose compilation time is calculated from python).

You're likely right. It couldn't hurt to make the same change for that benchmark.

Reply:
I agree. See #10